Modules/arraymodule.c

```diff
@@ -130,11 +149,76 @@ enum machine_format_code {
 
 #define array_Check(op, state) PyObject_TypeCheck(op, state->ArrayType)
 
+static arraydata *
+arraydata_alloc(Py_ssize_t size, int itemsize)
+{
+    if ((size_t) size > (PY_SSIZE_T_MAX - sizeof(arraydata))/itemsize - 1) {
```

Comment:
What is the `- 1` for here?

Reply:
extra safety margin I guess because originally signed 0x7fffffffffffffff couldn't be larger than anything, it is an impossible value anyway

Reply:
I still don't understand and I'm also a bit confused by the cast to `size_t` here.

We already have checks for size in the callers (around line 281 and 784). We should make those sufficient I think and remove this one.

Reply:
Its just a check that was originally there in mem alloc I think which is redundant now (or was redundant originally actually), will remove.